crates/warg/src/operator/mod.rs

```diff
@@ -0,0 +1,213 @@
+use prost::Message;
+use thiserror::Error;
+
+use crate::hash;
+
+pub mod model;
+pub mod validate;
+
+use crate::{protobuf, Encode, ErrorBox, Signable};
+
+// Deserialization
+
+impl TryFrom<&[u8]> for model::OperatorRecord {
+    type Error = ErrorBox;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        protobuf::OperatorRecord::decode(bytes)?.try_into()
+    }
+}
+
+impl TryFrom<protobuf::OperatorRecord> for model::OperatorRecord {
+    type Error = ErrorBox;
+
+    fn try_from(record: protobuf::OperatorRecord) -> Result<Self, Self::Error> {
+        let prev: Option<hash::Digest> = match record.prev {
+            Some(hash_string) => Some(hash_string.parse()?),
+            None => None,
+        };
+        let version = record.version;
+        let pbjson_timestamp = record.time.ok_or_else(|| Box::new(InvalidTimestampError))?;
```

Comment:
fwiw `anyhow::Context` lets you do `record.time.context(InvalidTimestampError)?`

Reply:
Sounds nice.